sdk/go/common/resource/properties_path.go

```diff
@@ -299,6 +301,184 @@ func (p PropertyPath) Contains(other PropertyPath) bool {
+		if key < 0 {
+			return false
+		}
+
+		// This is a leaf path element, so we want to reset the value at this index in new to the value at this index from old
+		if len(p) == 1 {
+			if !old.IsArray() && !new.IsArray() {
+				// Neither old nor new are arrays, so we can't reset this index
+				return true
+			} else if !old.IsArray() || !new.IsArray() {
+				// One of old or new is an array but the other isn't, so this is a path error
+				return false
+			}
+
+			// If neither array contains this index then this is a _same_ and so ok, e.g. given old:[1, 2] and
+			// new:[1] and a path of [3] we can return true because new at [3] is the same as old at [3], it
+			// doesn't exist.
+			if key >= len(old.ArrayValue()) && key >= len(new.ArrayValue()) {
+				return true
+			}
+			// If one array has this index but the other doesn't this is a path failure because we can't
+			// remove a location from an array.
+			if key >= len(old.ArrayValue()) || key >= len(new.ArrayValue()) {
+				return false
+			}
+			// Otherwise both arrays contain this index and we can reset the value of it in new to what is in
+			// old.
+			v := old.ArrayValue()[key]
+			new.ArrayValue()[key] = v
+			return true
+		}
+
+		if !old.IsArray() || !new.IsArray() {
+			// At least one of old or new is not an array, so we can't keep searching along this path but
+			// we only return an error if both are not arrays.
+			return !old.IsArray() && !new.IsArray()
+		}
+
+		// If this index is out of bounds in either array then this is a path failure because we can't
+		// continue the search of this path down each PropertyValue.
+		if key >= len(old.ArrayValue()) || key >= len(new.ArrayValue()) {
+			return false
+		}
+		old = old.ArrayValue()[key]
+		new = old.ArrayValue()[key]
+		return p[1:].reset(old, new)
+
+	case string:
+		if key == "*" {
+			if len(p) == 1 {
+				if new.IsObject() {
+					if old.IsObject() {
+						for k := range old.ObjectValue() {
+							v := old.ObjectValue()[k]
+							new.ObjectValue()[k] = v
+						}
+						for k := range new.ObjectValue() {
+							if _, has := old.ObjectValue()[k]; !has {
+								delete(new.ObjectValue(), k)
+							}
+						}
+					}
+					return true
+				} else if new.IsArray() {
+					if old.IsArray() {
+						for i := range old.ArrayValue() {
+							v := old.ArrayValue()[i]
+							new.ArrayValue()[i] = v
+						}
+					}
+					return true
+				} else {
+					return false
+				}
+			}
+
+			if old.IsObject() && new.IsObject() {
+				for k := range old.ObjectValue() {
+					var hasOld, hasNew bool
+					old, hasOld = old.ObjectValue()[k]
+					new, hasNew = new.ObjectValue()[k]
+					if !hasOld || !hasNew {
+						return false
+					}
+
+					if !p[1:].reset(old, new) {
+						return false
+					}
+				}
+				return true
+			} else if old.IsArray() && new.IsArray() {
+				for i := range old.ArrayValue() {
+					if !p[1:].reset(old.ArrayValue()[i], new.ArrayValue()[i]) {
+						return false
+					}
+				}
+				return true
```

Comment:
This bit is also not covered by unit tests. I think a non-leaf `[*]` is needed for this.